apisix/init.lua

```diff
@@ -358,7 +358,15 @@ local function verify_https_client(ctx)
         return true
     end
 
-    local matched_ssl = ngx.ctx.matched_ssl
+    local host = ctx.var.host
+    if not ctx.matched_ssl then
```

Comment:
Before that, `api_ctx.matched_ssl` was not initialized and the `api_ctx` will be destroyed after SSL handshake. 

You need to check `ngx.ctx.matched_ssl`, not `api_ctx.matched_ssl`.

Reply:
cc: @kingluo might explain better

Reply:
Yes, should use ngx.ctx.matched_ssl instead. It's my typo.

Reply:
However, it's a trivial stuff, because it only costs one more matching step.

Reply:
Agreed with @AlinsRan

Reply:
done

Reply:
tests fail with ngx.ctx for some reason

Reply:
Also, when changing on just this line https://github.com/apache/apisix/pull/10066/files#diff-7439ffe94cc4bcad12066d5d83f6c1004b8e9b3ff5097b645fa7c700eb3b3743R363, we get an error because after the if block we still try to access `matched_ssl` from `api_ctx` so I made changes below as well to extract from `ngx.ctx`